test/externalTests/exampleInventory.js

```diff
@@ -1,38 +1,83 @@
 const assert = require('assert')
 
+const tests = [
+  {
+    command: 'list',
+    wantedMessage: 'dirt x 64, stone x 64, iron_ore x 64, diamond_boots x 1'
+  },
+
+  {
+    command: 'equip hand dirt',
+    wantedMessage: 'equipped dirt'
+  },
+
+  {
+    command: 'toss 64 dirt',
+    wantedMessage: 'tossed 64 x dirt'
+  },
+
+  {
+    command: 'craft 16 stone_brick_stairs',
+    wantedMessage: 'I cannot make stone_brick_stairs'
+  },
+
+  {
+    command: 'craft 16 stone_bricks',
+    wantedMessage: 'did the recipe for stone_bricks 16 times'
+  },
+
+  {
+    command: 'craft 10 stone_brick_stairs',
+    wantedMessage: 'did the recipe for stick 10 stone_brick_stairs'
+  },
+
+  {
+    command: 'equip feet diamond_boots',
+    wantedMessage: 'equipped diamond_boots'
+  },
+
+  {
+    command: 'toss iron_ore',
+    wantedMessage: 'tossed iron_ore'
+  },
+
+  {
+    command: 'unequip feet',
+    wantedMessage: 'unequipped'
+  },
+
+  { // after tests layout
+    command: 'list',
+    wantedMessage: 'stone_bricks x 4, stone_brick_stairs x 40, diamond_boots x 1'
+  }
+]
+
 module.exports = () => (bot, done) => {
-  bot.test.runExample('examples/inventory.js', (name, cb) => {
-    assert.strictEqual(name, 'inventory')
+  bot.test.runExample('examples/inventory.js', (ign, cb) => {
+    assert.strictEqual(ign, 'inventory')
     bot.chat('/op inventory') // to counteract spawn protection
+    bot.chat('/setblock ~2 ~ ~2 crafting_table') // to make stone bricks stairs
     bot.chat('/give inventory dirt 64')
-    const inventoryTest = [
-      cb => {
-        bot.test.tellAndListen(name, 'list', (message) => {
-          if (!message.startsWith('dirt x 64')) {
-            assert.fail(`Unexpected message: ${message}`) // error
-          }
-          return true // stop listening
-        }, cb)
-      },
-      cb => {
-        bot.test.tellAndListen(name, 'equip hand dirt', (message) => {
-          if (!message.startsWith('equipped dirt')) {
-            assert.fail(`Unexpected message: ${message}`) // error
-          }
-          return true // stop listening
-        }, cb)
-      },
-      cb => {
-        bot.test.tellAndListen(name, 'toss 64 dirt', (message) => {
-          if (!message.startsWith('tossed 64 x dirt')) {
-            assert.fail(`Unexpected message: ${message}`) // error
-          }
-          return true // stop listening
-        }, cb)
-      }
-    ]
+    bot.chat('/give inventory stone 64')
+    bot.chat('/give inventory iron_ore 64')
+    bot.chat('/give inventory diamond_boots 1')
+    const inventoryTest = Object.values(tests).map(testData => generateTest(testData, cb))
+    // start test
     setTimeout(() => {
       bot.test.callbackChain(inventoryTest, cb)
     }, 2000)
+
+    function generateTest ({ command, wantedMessage }, cb) {
+      return () => bot.test.tellAndListen(ign, command, makeListener(wantedMessage), cb)
```

Comment:
The function it returns should receive a callback (which is passed by `bot.test.callbackChain()`), see for example https://github.com/PrismarineJS/mineflayer/blob/d792dbf44aeda53749e76c5689107492cead09d7/test/externalTests/bed.js#L13-L53